ignite/services/chain/generate.go

```diff
@@ -130,12 +130,8 @@ func (c *Chain) Generate(
 
 	// generate Typescript Client code as well if it is enabled.
 	if targetOptions.isTSClientEnabled {
-		tsClientPath := conf.Client.Typescript.Path
-		if tsClientPath == "" {
-			tsClientPath = defaultTSClientPath
-		}
-
-		tsClientRootPath := filepath.Join(c.app.Path, tsClientPath)
+		// TODO: Change to allow full paths in case TS client dir is not inside the app's dir?
```

Comment:
This option is interesting for developers that want to use that code for an other project. For instance they clone osmosis and want to generate the ts-client for their other project.

But osmosis has no config.yml, and even if it had, it's not convenient for the develop to update this file just to set the ts-client directory target. 

So this usecase should be handled by the `-path/-p` flag, but unfortunately, this flag is ignored by the `ts-client` command. Whatever the value passed to `-p`, the ts code is always generated in `ts-client` folder. Maybe you can fix that in this PR ?

Reply:
Implemented in 343c903 by adding an `--output/-o` flag because the `--path/-p` flag was already a global flag used for the app path.

I am not fully convinced with the implementation because it seems it doesn't follow the `Chain.Generate()` semantics for the target arguments. Now there is a new `TSClientPath()` "target" argument to override the configured typescript client path. Semantically that argument is not a target like ts-client, vuex, dart, and so on, so it seems off. Sending the _path_ as an argument to `GenerateTSClient()` could be an alternative but we would have to send an empty string if we would want to use the configured of default ts-client output path, so if that is preferred I can change it.

Reply:
Yes I see what you mean, I would go for your second proposal, adding the path in `GenerateTSClient()` func. I think that's acceptable, I've already seen such pattern in the codebase like in `cosmosclient.WithUseFaucet()` for instance.

The alternatives I have in mind : 

1) make path *optional*
```go
// GenerateTSClient enables generating proto based Typescript Client.
func GenerateTSClient(path ...string) GenerateTarget {
	return func(o *generateOptions) {
		o.isTSClientEnabled = true
		if len(path) > 0 {
			o.tsClientPath = path[0]
		}
	}
}
```

2) Using a struct (a little far-fetched ^^')

```go
// GenerateTSClient enables generating proto based Typescript Client.
type GenTsClient struct {
	path string
}

func (g GenTsClient) F(o *generateOptions) {
	o.isTSClientEnabled = true
	if g.path != "" {
		o.tsClientPath = g.path
	}
}

// (...)

err = c.Generate(cmd.Context(), cacheStorage, chain.GenerateTSClient{Path: output}.F)
```

But I think I prefer the second proposal :)

Reply:
Thank you very much for the feedback @tbruyelle!!